scripts/question_answering/model.py

```diff
@@ -0,0 +1,752 @@
+
+        Returns
+        --------
+        x : NDArray
+            output tensor with shape `(batch_size, sequence_length, hidden_size)`
+        mask : NDArray
+            output tensor with shape `(batch_size, sequence_length)`
+        """
+        x = self.position_encoder(x)
+        for conv in self.convs:
+            residual = x
+            x = conv(x) + residual
+        residual = x
+        x = self.attention_layer_norm(x)
+        x = F.Dropout(x, p=opt.layers_dropout)
+        x = self.attention(x, mask)
+        x = self.attention_dropout(x) + residual
+        return x + self.positionwise_ffn(x)
+
+
+class StochasticDropoutLayer(gluon.HybridBlock):
+    r"""
+    Stochastic dropout a layer.
+    """
+
+    def __init__(self, dropout, **kwargs):
+        super(StochasticDropoutLayer, self).__init__(**kwargs)
+        self.dropout = dropout
+        with self.name_scope():
+            self.dropout_fn = gluon.nn.Dropout(dropout)
+
+    def hybrid_forward(self, F, inputs):
+        if F.random.uniform().asscalar() < self.dropout:
+            return F.zeros(shape=(1,))
+        else:
+            return self.dropout_fn(inputs)
+
+
+class SelfAttention(gluon.HybridBlock):
+    r"""
+    Implementation of self-attention with gluonnlp.model.MultiHeadAttentionCell
+    """
+
+    def __init__(self, num_heads, **kwargs):
+        super(SelfAttention, self).__init__(**kwargs)
+        with self.name_scope():
+            self.attention = MultiHeadAttentionCell(
+                num_heads=num_heads,
+                base_cell=DotProductAttentionCell(
+                    scaled=True,
+                    dropout=opt.layers_dropout,
+                    use_bias=False
+                ),
+                query_units=opt.emb_encoder_conv_channels,
+                key_units=opt.emb_encoder_conv_channels,
+                value_units=opt.emb_encoder_conv_channels,
+                use_bias=False,
+                weight_initializer=Xavier()
+            )
+
+    def hybrid_forward(self, F, x, mask):
+        r"""Implement forward computation.
+
+        Parameters
+        -----------
+        x : NDArray
+            input tensor with shape `(batch_size, sequence_length, hidden_size)`
+        mask : NDArray
+            input tensor with shape `(batch_size, sequence_length)`
+
+        Returns
+        --------
+        x : NDArray
+            output tensor with shape `(batch_size, sequence_length, hidden_size)`
+        """
+        mask = F.batch_dot(mask.expand_dims(axis=2), mask.expand_dims(axis=1))
+        return self.attention(x, x, mask=mask)[0]
+
+
+class PositionEncoder(gluon.HybridBlock):
+    r"""
+    An implementation of position encoder.
+    """
+
+    def __init__(self, **kwargs):
+        super(PositionEncoder, self).__init__(**kwargs)
+        with self.name_scope():
+            pass
+
+    def hybrid_forward(self, F, x, min_timescale=1.0, max_timescale=1e4):

```

Comment:
I think setting python parameters in a `hybrid_forward` will not work. Did you try changing them? I believe the model will always use the value of the parameter used during the first forward pass. Therefore, please move `min_timescale=1.0, max_timescale=1e4` to the `__init__` function in case they need not be changed between different training iterations, or pass a 1D NDArray containing a single element instead of the Python floats